File: Minsk/SyntaxToken.cs
using System.Text;

namespace Minsk;

public record SyntaxToken
{
    public SyntaxKind Kind { get; }
    public string Text { get; }
    public int Position { get; }
    public object? Value { get; }

    public SyntaxToken(SyntaxKind kind, string text, int position, object? value)
    {
        Kind = kind;
        Text = text;
        Position = position;
        Value = value;
    }

    public override string ToString()
    {
        var result = new StringBuilder();
        result.Append($"{Kind} '{Text}'");
        if (Value != null)
        {
            result.Append($" {Value}");
        }
        return result.ToString();
    }
}

File: kt/Minsk/src/main/kotlin/dev/phytolizer/minsk/SyntaxToken.kt
package dev.phytolizer.minsk

class SyntaxToken(override val kind: SyntaxKind, val position: Int, val text: String, val value: Any?) : SyntaxNode() {
    override fun toString(): String {
        val result = StringBuilder()
        result.append("$kind '$text'")
        if (value != null) {
            result.append(" $value")
        }
        return result.toString()
    }

    override val children: List<SyntaxNode>
        get() = listOf()
}
